Add documentation for syntax-check rule (#2351)
ansible/ansible-lint: src/ansiblelint/rules/syntax_check.md

## syntax-check

> Our linter does run `ansible-playbook --syntax-check` on all playbooks, and

Reviewer: ```suggestion
Our linter runs `ansible-playbook --syntax-check` on all playbooks, and
```

if any of these reports a syntax error, this stops any further processing
of these files.

This error **cannot be disabled** due to being a prerequisite for other steps.

> You can either exclude these files from linting or better assure they can be

Reviewer: ```suggestion
You can exclude these files from linting, but it is better to make sure they can be
```

loaded by Ansible. This is often achieved by editing the inventory file and/or
`ansible.cfg` so ansible can load required variables.

> If undefined variables are the failure reason you could use the jinja
> `default()` filter for providing fallback values, like in the example below.

Reviewer: ```suggestion
If undefined variables cause the failure, you can use the jinja
`default()` filter to provide fallback values, like in the example below.
```

> This rule is among the few `unskippable` ones, rules that cannot be added

Reviewer: ```suggestion
This rule is among the few `unskippable` rules that cannot be added
```

to `ignore_list` or `warn_list`. One possible workaround is to add the entire
file to the `exclude_list`. This is a valid approach for special cases, like

> testing fixtures that are on-purpose invalid.

Reviewer: ```suggestion
testing fixtures that are invalid on purpose.
```
